src/chocolatey/infrastructure.app/nuget/NugetCommon.cs

```diff
@@ -451,16 +451,27 @@ ISet<SourcePackageDependencyInfo> dependencyInfos
 
             //if (availablePackages.Contains(packageID)) return;
 
-            var dependencyInfoResources = resources.DependencyInfoResources();
-
-            foreach (var dependencyInfoResource in dependencyInfoResources)
+            foreach (var resource in resources)
             {
-                IEnumerable<SourcePackageDependencyInfo> dependencyInfos = Array.Empty<SourcePackageDependencyInfo>();
+                var dependencyInfoResource = resource.DependencyInfoResource;
+
+                if (dependencyInfoResource is null)
+                {
+                    // We can't lookup any dependencies using this resource.
+                    continue;
+                }
+
+                // check if we already have packages matching the constraints in the list
+                IEnumerable<SourcePackageDependencyInfo> dependencyInfos = availablePackages.Where(
+                    p => string.Equals(p.Id, packageId, StringComparison.OrdinalIgnoreCase) && p.HasVersion && versionRange.Satisfies(p.Version) && p.Source == resource.Source).ToList();
 
                 try
                 {
-                    dependencyInfos = await dependencyInfoResource.ResolvePackages(
-                        packageId, configuration.Prerelease, framework, cacheContext, logger, CancellationToken.None);
+                    if (!dependencyInfos.Any())

```

Comment:
I believe the `Any()` used here will cause unnecessary enumeration. We should be able to use the `Count` property of the `List<T>` type.

Reply:
I see that we're using `.Any()` down on line 485. It makes sense then to me for us to use it here for that consistency.